hudi-utilities/src/main/java/org/apache/hudi/utilities/keygen/TimestampBasedKeyGenerator.java

```diff
@@ -96,8 +95,7 @@ public HoodieKey getKey(GenericRecord record) {
       } else if (partitionVal instanceof String) {
         unixTime = inputDateFormat.parse(partitionVal.toString()).getTime() / 1000;
       } else {
-        throw new HoodieNotSupportedException(
-            "Unexpected type for partition field: " + partitionVal.getClass().getName());
+        unixTime = 1L;
```

Comment:
I dont think we should use default for this. I can understand using default when the value is null. But here, it is of unknown type. This could very well be because of some user configuration error ? Is there any specific reason why we should default to 1 instead of erroring out ?

Reply:
@bvaradar I changed here with the idea that when partitionVal is returned as null from DataSourceUtils, it will not match any of the instanceof conditions, and will result in throwing the exception according to previous code. 
But this is a valid point that the exception can be due to some user configuration error as well. Will modify the code accordingly.